sklearn/ensemble/tests/test_iforest.py

```diff
@@ -0,0 +1,147 @@
+
+"""
+Testing for Isolation Forest algorithm (sklearn.ensemble.iforest).
+"""
+
+# Authors: Nicolas Goix <nicolas.goix@telecom-paristech.fr>
+#          Alexandre Gramfort <alexandre.gramfort@telecom-paristech.fr>
+# License: BSD 3 clause
+
+import numpy as np
+
+from sklearn.utils.testing import assert_array_equal
+from sklearn.utils.testing import assert_array_almost_equal
+from sklearn.utils.testing import assert_raises
+from sklearn.utils.testing import assert_warns
+from sklearn.utils.testing import assert_equal
+from sklearn.utils.testing import assert_true
+from sklearn.utils.testing import ignore_warnings
+
+from sklearn.grid_search import GridSearchCV, ParameterGrid
+from sklearn.ensemble import IsolationForest
+from sklearn.cross_validation import train_test_split
+from sklearn.datasets import load_boston, load_iris
+from sklearn.utils import check_random_state
+
+from scipy.sparse import csc_matrix, csr_matrix
+
+rng = check_random_state(0)
+
+# load the iris dataset
+# and randomly permute it
+iris = load_iris()
+perm = rng.permutation(iris.target.size)
+iris.data = iris.data[perm]
+iris.target = iris.target[perm]
+
+# also load the boston dataset
+# and randomly permute it
+boston = load_boston()
+perm = rng.permutation(boston.target.size)
+boston.data = boston.data[perm]
+boston.target = boston.target[perm]
+
+
```

Comment:
Yes I know. But there are no tests for that. In my opinion, we should at least test that outliers are indeed scored as "abnormal". (Said otherwise, we should test that the implementation does what it says)